### ISSUE TITLE
Support Iceberg sort transforms

### ISSUE THREAD

**Issue by evanvdia** [prestodb/presto#24250]
Follow up from:https://github.com/prestodb/presto/pull/21977
The Iceberg specification supports transforms on sort_by columns. For example truncate(col_0, 5) to sort based on the first 5 characters in col_0.

**Comment by PingLiuPing** [comment]
Is this supported now? Thanks.

**Comment by hantangwangd** [comment]
As far as I know, it hasn't been implemented yet.

**Comment by PingLiuPing** [comment]
@hantangwangd  thank you. 
 Does Presto implement the sorting based on original values of each column? According to https://iceberg.apache.org/spec/#sorting, we can assume it implements sorting based on identity transform.
But other than identity and truncate transforms, do we need to implement sorting based on other transform's result?

**Comment by hantangwangd** [comment]
@PingLiuPing Yes, PR #21977 has implemented sorting based on original values (identity transform) for Iceberg tables. But we haven't supported any non-identity sort transform yet, see the documentation [here](https://github.com/prestodb/presto/blob/master/presto-docs/src/main/sphinx/connector/iceberg.rst?plain=1#L2162-L2221).
